src/main/java/com/notably/model/block/BlockNodeImpl.java

```diff
@@ -0,0 +1,145 @@
+package com.notably.model.block;
+
+import static java.util.Objects.hash;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import com.notably.model.block.exceptions.DuplicateBlockException;
+import com.notably.model.block.exceptions.NoSuchBlockException;
+
+import javafx.collections.ObservableList;
+import javafx.scene.control.TreeItem;
+
+/**
+ * Represents a BlockNode in the Notably data structure.
+ */
+public class BlockNodeImpl extends TreeItem<Block> implements BlockNode {
+    private TreeItem<Block> blockNode;
+
+    public BlockNodeImpl(Block block) {
+        requireNonNull(block);
+        blockNode = new TreeItem<Block>(block);
+    }
+
+    public BlockNodeImpl(TreeItem<Block> treeItem) {
+        requireNonNull(treeItem);
+        blockNode = treeItem;
+    }
+
+    public static BlockNode createRootBlockNode() {
+        return new BlockNodeImpl(BlockImpl.createRootBlock());
+    }
+
+    public static BlockNode createBlockNode(Block block) {
+        return new BlockNodeImpl(block);
+    }
```

Comment:
Why do we need this method? Isn't it the same as one of the constructors?

Reply:
Fixed along with other changes. Please advise!